src/nemo_evaluator/solvers/harbor.py

```
@@ -1374,6 +1375,54 @@ def _patch_terminus_api_token_anchor() -> None:
+#
+# Harbor v0.18.0 added exactly this substring upstream (see
+# ``harbor/llms/lite_llm.py`` with the comment ``# vllm 0.16.0 error
+# message``), so this patch becomes redundant the moment the ``harbor`` pin
+# in ``pyproject.toml`` is bumped to a release containing it.
+_VLLM_CTXLIM_PHRASES: tuple[str, ...] = (
+    "model's context length",  # vLLM VLLMValidationError; harbor >=0.18 catches this natively
+)
+
+
+def _patch_harbor_lite_llm_context_length_matcher() -> None:
+    global _HARBOR_LITELLM_CTXLIM_PATCHED
```

Ruff (0.15.20)

[warning] Using the global statement to update `_HARBOR_LITELLM_CTXLIM_PATCHED` is discouraged

(PLW0603)